cpp/src/gandiva/expr_validator.cc

```diff
@@ -154,4 +154,67 @@ Status ExprValidator::Visit(const BooleanNode& node) {
   return Status::OK();
 }
 
+/*
+ * Validate the following
+ *
+ * 1. Non empty list of constants to search in.
+ * 2. Expression returns of the same type as the constants.
+ */
+Status ExprValidator::Visit(const InExpressionNode<int32_t>& node) {
```

Comment:
The code duplication in these three overloads looks gratuitous to me. Can't you factor them out?

Reply:
Yes I should!